### PR TITLE
Fix exceptions not being caught

### LINKED ISSUE #1297: Tests fail on ppc64le
As reported in #922.

### PR DESCRIPTION
**Pull request by brunoalr**
Adding -fexceptions to both torch and pytorch C/C++ builds fixes tests
not passing in ppc64le and aarch64.

See https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_exceptions.html

Closes #1297

### CODE REVIEW

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
The PR breaks CUDA builds (nvcc doesn't support `-fexceptions`)

**Comment by brunoalr** [comment]
Right now I can't do a CUDA build. But the main idea should be just not to pass `-fexceptions` to nvcc in torch/lib. I will take a look at it.

**Comment by soumith** [comment]
from what i get, you only need `-fexceptions` in setup.py. The other change is not needed. It's because we make our TH error handler to throw exceptions on the python extension side at runtime.
That would also fix the CUDA issue.

**Comment by brunoalr** [comment]
@soumith I tested giving up the changes in build_all.sh and unfortunately that does not work. We have to pass the flag to both C and C++ builds.

**Comment by apaszke** [comment]
@soumith The flag is only needed in the files compiled with the C compiler, so it's needed for `torch/lib` too.

**Comment by brunoalr** [comment]
@apaszke @soumith can you review this updated PR, please?

File: torch/lib/build_all.sh [hunk 1]
```diff
@@ -36,14 +36,20 @@ else
     LDFLAGS="$LDFLAGS -Wl,-rpath,\$ORIGIN"
 fi
 C_FLAGS="$BASIC_C_FLAGS $LDFLAGS"
+NVCC_TARGETS="THC THCS THCUNN"
 function build() {
   mkdir -p build/$1
   cd build/$1
+  BUILD_C_FLAGS=''
+  eval 'case '$1' in
+      '${NVCC_TARGETS//" "/" | "}') BUILD_C_FLAGS=$C_FLAGS;;
```
Comment (apaszke) [inline torch/lib/build_all.sh]: Have you verified that it still fixes it on PPC? That flag might be necessary when compiling `.c` files in THC.

Comment (brunoalr) [inline torch/lib/build_all.sh]: It fixes ppc builds without CUDA. I finally got access to a CUDA-capable machine and let the tests run overnight. ppc CUDA build is still broken. I'll look into that right now and ping you later. Thanks for point this out.

**Comment by soumith** [comment]
@brunoalr i presume you didn't get time after that to work on this. I'm closing the PR for now, it's quite stale.

**Comment by brunoalr** [comment]
@soumith sorry, it took me more time than expected to get access to a machine with a GPU. I tested the updated patch and it's working.

I opened a new PR: https://github.com/pytorch/pytorch/issues/1948
Please, take a look.

**Pull request closed without merging**